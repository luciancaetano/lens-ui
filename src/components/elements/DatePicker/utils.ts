import { DateObject, toDateObject } from 'react-multi-date-picker';
import { DatePickerType } from './DatePicker.types';

export function toDate(value: Date): DateObject {
  const dt = toDateObject(value);
  return dt.isValid ? dt : toDateObject(new Date());
}

export function parseValue(value: Date | number | string | Date[] | [Date, Date], type: DatePickerType) {
  if (Array.isArray(value) && type === 'range') {
    return value.length === 0 ? [toDateObject(new Date()), toDateObject(new Date())] : value.map((v: Date) => toDate(v));
  }

  if (!value && type === 'range') {
    return [toDateObject(new Date()), toDateObject(new Date())];
  }

  if (typeof value === 'number' && type === 'month') {
    const dt = toDateObject(new Date());
    dt.setMonth(value);
    return dt;
  }

  if (!value && type === 'month') {
    return toDateObject(new Date());
  }

  if (typeof value === 'number' && type === 'year') {
    const dt = toDateObject(new Date());
    dt.setYear(value);
    return dt;
  }

  if (!value && type === 'year') {
    return toDateObject(new Date());
  }

  if (!Array.isArray(value) && typeof value === 'string' && type === 'time') {
    return toDateObject(new Date(value));// todo handle time
  }

  if (!Array.isArray(value) && type === 'date') {
    return toDate(value as Date);
  }

  return toDateObject(new Date());
}

export function init(value: Date | number | string | Date[] | [Date, Date], defaultValue: Date | number | string | Date[] | [Date, Date], type: DatePickerType) {
  if (value === undefined) {
    return parseValue(defaultValue, type);
  }

  return parseValue(value, type);
}
