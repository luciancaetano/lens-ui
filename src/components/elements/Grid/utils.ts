import { GridBreakpointsType } from './Grid.types';

const bpSortOrder = ['xs', 'sm', 'md', 'lg'];

export function sortBreakpoints(a: GridBreakpointsType, b: GridBreakpointsType) {
  return bpSortOrder.indexOf(a) - bpSortOrder.indexOf(b);
}
