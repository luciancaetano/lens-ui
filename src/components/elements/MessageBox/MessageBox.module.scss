@import '../../../style-lib';

.message-box {
  @include lens-ui-typography;

  --stripped-opacity: .03;

  display: flex;
  overflow: hidden;
  font-size: .65625rem;
  border-radius: .1rem;
  flex-direction: column;
  position: relative;
  margin-top: .2rem;
  margin-bottom: .2rem;
  box-sizing: border-box;
  transition: opacity linear .3s;
  box-shadow: var(--lens-ui-elements-card-box-shadow);

  &--hidden {
    opacity: 0;
  }

  &__title {
    @include lens-ui-typography-no-color;

    padding: .2rem;
    flex: 1;
    font-size: .84rem;
    font-weight: bold;
  }

  &__content {
    padding: .2rem;
    flex: 1;
    font-size: .8rem;
  }

  [data-lens-element='progress-bar'],
  [data-lens-element='progress-bar__indicator'] {
    border-radius: 0;
    height: .3rem !important;
  }

  &__close-button {
    position: absolute;
    border: 0;
    background: transparent;
    top: 0;
    right: 0;
    display: flex;
    width: 20px;
    height: 20px;
    padding: 0;
    align-items: center;
    justify-content: center;
    cursor: pointer;
  }

  &--striped {
    background-image: linear-gradient(45deg, rgb(255 255 255 / var(--stripped-opacity)) 25%, transparent 25%, transparent 50%, rgb(255 255 255 / var(--stripped-opacity)) 50%, rgb(255 255 255 / var(--stripped-opacity)) 75%, transparent 75%, transparent);
    background-size: .625rem .625rem;
  }

  &--intent-primary {
    background-color: var(--lens-ui-intents-primary);
    color: var(--lens-ui-intents-primary-text);
    border: 1px solid var(--lens-ui-intents-primary-border);

    [data-lens-element='message-box__close-button'],
    [data-lens-element='message-box__close-button'] > svg {
      fill: var(--lens-ui-intents-primary-text);
    }

    [data-lens-element='toast-close-button'] {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-primary-dark-text);
    }

    [data-lens-element='progress-bar'] {
      background-color: var(--lens-ui-intents-primary-darken);
    }

    [data-lens-element='progress-bar__indicator'] {
      background-color: #fff;
    }
  }

  &--intent-secondary {
    background-color: var(--lens-ui-intents-secondary);
    color: var(--lens-ui-intents-secondary-text);
    border: 1px solid var(--lens-ui-intents-secondary-border);

    [data-lens-element='message-box__close-button'],
    [data-lens-element='message-box__close-button'] > svg {
      fill: var(--lens-ui-intents-secondary-text);
    }

    [data-lens-element='toast-close-button'] {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-secondary-dark-text);
    }

    [data-lens-element='progress-bar'] {
      background-color: var(--lens-ui-intents-secondary-darken);
    }

    [data-lens-element='progress-bar__indicator'] {
      background-color: #fff;
    }
  }

  &--intent-success {
    background-color: var(--lens-ui-intents-success);
    color: var(--lens-ui-intents-success-text);
    border: 1px solid var(--lens-ui-intents-success-border);

    [data-lens-element='message-box__close-button'],
    [data-lens-element='message-box__close-button'] > svg {
      fill: var(--lens-ui-intents-success-text);
    }

    [data-lens-element='toast-close-button'] {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-success-dark-text);
    }

    [data-lens-element='progress-bar'] {
      background-color: var(--lens-ui-intents-success-darken);
    }

    [data-lens-element='progress-bar__indicator'] {
      background-color: #fff;
    }
  }

  &--intent-info {
    background-color: var(--lens-ui-intents-info);
    color: var(--lens-ui-intents-info-text);
    border: 1px solid var(--lens-ui-intents-info-border);

    [data-lens-element='message-box__close-button'],
    [data-lens-element='message-box__close-button'] > svg {
      fill: var(--lens-ui-intents-info-text);
    }

    [data-lens-element='toast-close-button'] {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-info-dark-text);
    }

    [data-lens-element='progress-bar'] {
      background-color: var(--lens-ui-intents-info-darken);
    }

    [data-lens-element='progress-bar__indicator'] {
      background-color: #fff;
    }
  }

  &--intent-warning {
    background-color: var(--lens-ui-intents-warning);
    color: var(--lens-ui-intents-warning-text);
    border: 1px solid var(--lens-ui-intents-warning-border);

    [data-lens-element='message-box__close-button'],
    [data-lens-element='message-box__close-button'] > svg {
      fill: var(--lens-ui-intents-warning-text);
    }

    [data-lens-element='toast-close-button'] {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-warning-dark-text);
    }

    [data-lens-element='progress-bar'] {
      background-color: var(--lens-ui-intents-warning-darken);
    }

    [data-lens-element='progress-bar__indicator'] {
      background-color: #fff;
    }
  }

  &--intent-danger {
    background-color: var(--lens-ui-intents-danger);
    color: var(--lens-ui-intents-danger-text);
    border: 1px solid var(--lens-ui-intents-danger-border);

    [data-lens-element='message-box__close-button'],
    [data-lens-element='message-box__close-button'] > svg {
      fill: var(--lens-ui-intents-danger-text);
    }

    [data-lens-element='toast-close-button'] {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-danger-dark-text);
    }

    [data-lens-element='progress-bar'] {
      background-color: var(--lens-ui-intents-danger-darken);
    }

    [data-lens-element='progress-bar__indicator'] {
      background-color: #fff;
    }
  }
}
