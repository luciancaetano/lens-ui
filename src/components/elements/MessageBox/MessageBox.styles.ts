import styled from 'styled-components';

export const Container = styled.div`
  --stripped-opacity: .03;

  display: flex;
  overflow: hidden;
  font-size: .65625rem;
  border-radius: .1rem;
  flex-direction: column;
  position: relative;
  margin-top: .2rem;
  margin-bottom: .2rem;
  box-sizing: border-box;
  transition: opacity linear .3s;

  &.hidden {
    opacity: 0;
  }

  .lens-ui-message-box-title {
    padding: .2rem;
    flex: 1;
    font-size: .8rem;
    font-weight: bold;
  }

  .lens-ui-message-box-content {
    padding: .2rem;
    flex: 1;
    font-size: .8rem;
  }

  .lens-ui-progress-bar,
  .lens-ui-progress-bar-progress {
    border-radius: 0;
    height: .3rem;
  }

  .lens-ui-message-box-close-button {
    position: absolute;
    border: 0;
    background: transparent;
    top: 0;
    right: 0;
    display: flex;
    width: 20px;
    height: 20px;
    padding: 0;
    align-items: center;
    justify-content: center;
  }

  &.striped {
    background-image: linear-gradient(45deg, rgba(255,255,255,var(--stripped-opacity)) 25%,transparent 25%,transparent 50%,rgba(255,255,255,var(--stripped-opacity)) 50%,rgba(255,255,255,var(--stripped-opacity)) 75%,transparent 75%,transparent);
    background-size: .625rem .625rem;
  }

  &.intent-primary {
    background-color: var(--lens-ui-intents-primary);
    color: var(--lens-ui-intents-primary-text);
    border: 1px solid var(--lens-ui-intents-primary-border);

    .lens-ui-message-box-close-button {
      color: var(--lens-ui-intents-primary-text);
    }

    .toast-close-button {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-primary-dark-text);
    }

    .lens-ui-progress-bar {
      background-color: var(--lens-ui-intents-primary-darken);
    }

    .lens-ui-progress-bar-progress {
      background-color: #fff;
    }
  }

  &.intent-secondary {
    background-color: var(--lens-ui-intents-secondary);
    color: var(--lens-ui-intents-secondary-text);
    border: 1px solid var(--lens-ui-intents-secondary-border);

    .lens-ui-message-box-close-button {
      color: var(--lens-ui-secondary-primary-text);
    }

    .toast-close-button {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-secondary-dark-text);
    }

    .lens-ui-progress-bar {
      background-color: var(--lens-ui-intents-secondary-darken);
    }

    .lens-ui-progress-bar-progress {
      background-color: #fff;
    }
  }

  &.intent-success {
    background-color: var(--lens-ui-intents-success);
    color: var(--lens-ui-intents-success-text);
    border: 1px solid var(--lens-ui-intents-success-border);

    .lens-ui-message-box-close-button {
      color: var(--lens-ui-success-primary-text);
    }

    .toast-close-button {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-success-dark-text);
    }

    .lens-ui-progress-bar {
      background-color: var(--lens-ui-intents-success-darken);
    }

    .lens-ui-progress-bar-progress {
      background-color: #fff;
    }
  }

  &.intent-info {
    background-color: var(--lens-ui-intents-info);
    color: var(--lens-ui-intents-info-text);
    border: 1px solid var(--lens-ui-intents-info-border);

    .lens-ui-message-box-close-button {
      color: var(--lens-ui-success-info-text);
    }

    .toast-close-button {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-info-dark-text);
    }

    .lens-ui-progress-bar {
      background-color: var(--lens-ui-intents-info-darken);
    }

    .lens-ui-progress-bar-progress {
      background-color: #fff;
    }
  }

  &.intent-warning {
    background-color: var(--lens-ui-intents-warning);
    color: var(--lens-ui-intents-warning-text);
    border: 1px solid var(--lens-ui-intents-warning-border);

    .lens-ui-message-box-close-button {
      color: var(--lens-ui-success-warning-text);
    }

    .toast-close-button {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-warning-dark-text);
    }

    .lens-ui-progress-bar {
      background-color: var(--lens-ui-intents-warning-darken);
    }

    .lens-ui-progress-bar-progress {
      background-color: #fff;
    }
  }

  &.intent-danger {
    background-color: var(--lens-ui-intents-danger);
    color: var(--lens-ui-intents-danger-text);
    border: 1px solid var(--lens-ui-intents-danger-border);

    .lens-ui-message-box-close-button {
      color: var(--lens-ui-success-danger-text);
    }

    .toast-close-button {
      background-color: transparent;
      border: 0;
      color: var(--lens-ui-intents-danger-dark-text);
    }

    .lens-ui-progress-bar {
      background-color: var(--lens-ui-intents-danger-darken);
    }

    .lens-ui-progress-bar-progress {
      background-color: #fff;
    }
  }
`;
