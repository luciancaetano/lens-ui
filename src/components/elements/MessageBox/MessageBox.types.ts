import React from 'react';
import {
  IntentType, IPropsWithClassName, IPropsWithId, ITestableProps,
} from '../../../types';

export interface IMessageBoxProps extends ITestableProps, IPropsWithClassName, IPropsWithId {
  intent?: IntentType;
  icon?: React.ReactNode;
  title?: React.ReactNode;
  children: React.ReactNode;
  timeout?: number;
  striped?: boolean;
  onClose?: (isOpen: false) => void;
  closable?: boolean;
  isOpen?: boolean;
}
