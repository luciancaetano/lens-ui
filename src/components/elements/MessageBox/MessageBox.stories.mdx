import { Canvas, Meta, Story } from '@storybook/addon-docs';
import MessageBox from './MessageBox';

<Meta title="Components/MessageBox" component={MessageBox}/>


# MessageBox

MessageBox alert displays a short, important message in a way that attracts the user's attention without interrupting the user's task.

<Canvas>
  <MessageBox
    intent="primary"
    title="Title"
  >
    Hello world
  </MessageBox>
</Canvas>

### Properties

|Property|Type|Default Value|Description|Required|
|---|---|---|---|---|
|intent|`primary`, `secondary`, `success`, <br/>`info`, `warning`, `danger`|primary | The intent of the badge.|No|
|icon|`ReactNode`|-|The icon to display.|No|
|title|`ReactNode`|-|The title of the message box.|No|
|children|`ReactNode`|-|The content of the message box.|Yes|
|timeout|`number`|3000|The time in milliseconds to show the message box.|No|
|striped|`boolean`|false|Whether the message box is striped.|No|
|onClose|`(isOpen: false) => void`|-|The callback function to be called when the message box is closed.|No|
|closable|`boolean`|true|Whether the message box is closable.|No|
|isOpen|`boolean`|true|Whether the message box is open.|No|
|testingID| string | - | The id used for testing purposes.<br/>`<div data-testid="my-test-id"/>` |No|

---

### With timeout

<Canvas>
  <MessageBox
    intent="primary"
    title="Title"
    timeout={5000}
  >
    Hello world
  </MessageBox>
</Canvas>