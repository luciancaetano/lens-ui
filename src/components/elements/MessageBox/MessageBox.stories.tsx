/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import MessageBox from './MessageBox';
import { IMessageBoxProps } from './MessageBox.types';
import LensProvider from '../../providers/LensProvider/LensProvider';
import '../../../styles';
import { Intents } from '../../../types';

export default {
  title: '2. Components/MessageBox',
  component: MessageBox,
  decorators: [
    (Story) => <LensProvider>{Story()}</LensProvider>,
  ],
} as ComponentMeta<typeof MessageBox>;

const Template: ComponentStory<typeof MessageBox> = (args) => <MessageBox {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Hello world',
  intent: 'info',
} as IMessageBoxProps;

export const _MessageBoxStripped = Template.bind({});
_MessageBoxStripped.args = {
  children: 'Hello world',
  intent: 'info',
  title: 'title',
  striped: true,
} as IMessageBoxProps;

export const _Timeout = Template.bind({});
_Timeout.args = {
  children: 'Hello world',
  intent: 'info',
  title: 'title',
  striped: true,
  timeout: 3000,
} as IMessageBoxProps;

export const AllIntents = (args) => Object.keys(Intents).map((intent: any) => (
  <Template intent={intent} {...args} />
));

AllIntents.args = {
  title: 'Title',
  children: 'Hello world',
};
