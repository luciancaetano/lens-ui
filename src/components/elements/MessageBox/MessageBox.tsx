/* eslint-disable jsx-a11y/control-has-associated-label */
import clsx from 'clsx';
import React, {
  useRef, useState, useCallback, useEffect,
  useMemo,
} from 'react';
import styles from './MessageBox.module.scss';
import { IMessageBoxProps } from './MessageBox.types';
import ProgressBar from '../ProgressBar/ProgressBar';
import Icon from '../Icon/Icon';
import { useProgressiveTimeout } from '../../../hooks';
import { sleep } from '../../../utils';

const STOP_RENDER_TIMEOUT = 500;

const MessageBox: React.FC<IMessageBoxProps> = ({
  className, testingID, id, intent = 'primary', striped, title, children, timeout, icon, closable = true, onClose, isOpen,
  ...props
}) => {
  const [isOpenState, setIsOpen] = useState(true);
  const [renderDisabled, setRenderDisabled] = useState(false);
  const progressBar = useRef<HTMLDivElement>(null);
  const progressDone = useRef(false);
  const timeOutRef = useRef<Function>(null);

  const renderDisableTimeout = useRef(async () => {
    if (!timeOutRef.current) {
      await sleep(STOP_RENDER_TIMEOUT, timeOutRef);
      setRenderDisabled(true);
    }
  });

  const show = useMemo(() => isOpen !== undefined ? isOpen : isOpenState, [isOpenState, isOpen]);

  const updateProgressBar = useCallback((progress: number) => {
    if (!progressBar.current) return;
    const element = progressBar.current.querySelector<HTMLDivElement>('[data-lens-element=\'progress-bar__indicator\']');
    if (element) {
      element.style.width = `${progress}%`;
    }
  }, []);

  const [start, stopProgress, inProgress] = useProgressiveTimeout(updateProgressBar, () => {
    if (onClose) {
      onClose(false);
    }
    setIsOpen(false);
    renderDisableTimeout.current();
  });

  const handleClose = useCallback(() => {
    stopProgress();
    setIsOpen(false);
    updateProgressBar(100);
    if (onClose) {
      onClose(false);
    }
    renderDisableTimeout.current();
  }, [stopProgress, onClose, updateProgressBar]);

  useEffect(() => {
    if (timeout > 0 && !inProgress.current && !progressDone.current) {
      progressDone.current = true;
      start(timeout);
    }
  }, [timeout, start, inProgress]);

  useEffect(() => () => {
    if (timeOutRef.current) {
      timeOutRef.current();
    }
  }, []);

  if (renderDisabled && !show) return null;

  return (
    <div
      {...props}
      id={id}
      data-testid={testingID}
      data-lens-element="message-box"
      className={clsx(
        styles.messageBox,
        striped && styles.messageBoxStriped,
        !show && styles.messageBoxHidden,
        styles[`message-box--intent-${intent}`],
        className,
      )}
    >
      {title && <div data-lens-element="message-box__title" className={styles.messageBoxTitle}>{title}</div>}
      {closable && (
        <button
          data-lens-element="message-box__close-button"
          className={clsx(styles.messageBoxCloseButton, styles[`message-box--intent-${intent}__close-button`])}
          onClick={handleClose}
        >
          <Icon name="BsXSquareFill" size="0.8rem" />
        </button>
      )}
      <div>
        {icon && <div data-lens-element="message-box__icon">{icon}</div>}
        <div className={clsx(styles.messageBoxContent)}>{children}</div>
      </div>
      {timeout && <ProgressBar progress={0} ref={progressBar} />}
    </div>
  );
};

export default MessageBox;
