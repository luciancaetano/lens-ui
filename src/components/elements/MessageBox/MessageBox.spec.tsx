import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import MessageBox from './MessageBox';
import LensProvider from '../../providers/LensProvider/LensProvider';
import { sleep } from '../../../utils';

jest.useFakeTimers();

jest.mock('../../../utils', () => ({
  sleep: () => true,
  Layers: {
    Toast: 1,
  },
}));

describe('<MessageBox/>', () => {
  it('should test component rederization', async () => {
    const children = 'Hello world';
    const title = 'my_title';
    const testingId = 'my_testing_id';

    const { getByTestId, getByText } = render(
      <LensProvider>
        <MessageBox title={title} testingID={testingId}>
          {children}
        </MessageBox>
      </LensProvider>,
    );

    expect(getByTestId(testingId)).toBeInTheDocument();
    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(children)).toBeInTheDocument();
  });

  it('should test the close button uncontrolled state', async () => {
    const children = 'Hello world';
    const title = 'my_title';
    const testingId = 'my_testing_id';

    const { getByTestId, getByText, container } = render(
      <LensProvider>
        <MessageBox title={title} testingID={testingId} closable>
          {children}
        </MessageBox>
      </LensProvider>,
    );

    const childrenEl = getByText(children);

    expect(getByTestId(testingId)).toBeInTheDocument();
    expect(getByText(title)).toBeInTheDocument();
    expect(childrenEl).toBeInTheDocument();
    expect(container.querySelector('[data-lens-element="message-box__close-button"]')).toBeInTheDocument();

    fireEvent.click(container.querySelector('[data-lens-element="message-box__close-button"]'));

    jest.advanceTimersByTime(10000);

    await sleep(1000);

    expect(childrenEl).not.toBeInTheDocument();
    expect(container.querySelector('[data-lens-element="message-box__close-button"]')).not.toBeInTheDocument();
  });

  it('should test automatic closing', async () => {
    const children = 'Hello world';
    const title = 'my_title';
    const testingId = 'my_testing_id';
    const onClose = jest.fn();

    const { getByTestId, getByText, container } = render(
      <LensProvider>
        <MessageBox title={title} testingID={testingId} timeout={3000} onClose={onClose} icon={<>Icon</>} striped>
          {children}
        </MessageBox>
      </LensProvider>,
    );

    const childrenEl = getByText(children);

    expect(getByTestId(testingId)).toBeInTheDocument();
    expect(getByText(title)).toBeInTheDocument();
    expect(childrenEl).toBeInTheDocument();
    expect(container.querySelector('[data-lens-element="message-box__close-button"]')).toBeInTheDocument();

    jest.advanceTimersByTime(10000);

    await sleep(1000);

    expect(onClose).toBeCalled();
    expect(childrenEl).not.toBeInTheDocument();
    expect(container.querySelector('[data-lens-element="message-box__close-button"]')).not.toBeInTheDocument();
  });
});
