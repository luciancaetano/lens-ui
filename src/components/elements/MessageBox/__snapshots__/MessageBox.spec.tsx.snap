// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageBox/> should test automatic closing 1`] = `
<div>
  <div
    id="lens-ui-portal-modal-root"
  />
  <div
    class="messageBox messageBoxStriped message-box--intent-primary"
    data-lens-element="message-box"
    data-testid="my_testing_id"
  >
    <div
      class="messageBoxTitle"
      data-lens-element="message-box__title"
    >
      my_title
    </div>
    <button
      class="messageBoxCloseButton message-box--intent-primary__close-button"
      data-lens-element="message-box__close-button"
    >
      <svg
        class="icon"
        data-lens-element="icon"
        fill="currentColor"
        height="0.8rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="0.8rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm3.354 4.646L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 1 1 .708-.708z"
        />
      </svg>
    </button>
    <div>
      <div
        data-lens-element="message-box__icon"
      >
        Icon
      </div>
      <div
        class="messageBoxContent"
      >
        Hello world
      </div>
    </div>
    <div
      class="progressBar"
      data-lens-element="progress-bar"
      data-lens-intent="primary"
      style="height: 0.625rem;"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="progress-bar__indicator-intent-primary"
        data-lens-element="progress-bar__indicator"
        data-lens-intent="primary"
        role="progressbar"
        style="width: 0%;"
      />
    </div>
  </div>
</div>
`;

exports[`<MessageBox/> should test automatic closing 2`] = `
<div>
  <div
    id="lens-ui-portal-modal-root"
  />
  <div
    class="messageBox messageBoxStriped message-box--intent-primary"
    data-lens-element="message-box"
    data-testid="my_testing_id"
  >
    <div
      class="messageBoxTitle"
      data-lens-element="message-box__title"
    >
      my_title
    </div>
    <button
      class="messageBoxCloseButton message-box--intent-primary__close-button"
      data-lens-element="message-box__close-button"
    >
      <svg
        class="icon"
        data-lens-element="icon"
        fill="currentColor"
        height="0.8rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="0.8rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm3.354 4.646L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 1 1 .708-.708z"
        />
      </svg>
    </button>
    <div>
      <div
        data-lens-element="message-box__icon"
      >
        Icon
      </div>
      <div
        class="messageBoxContent"
      >
        Hello world
      </div>
    </div>
    <div
      class="progressBar"
      data-lens-element="progress-bar"
      data-lens-intent="primary"
      style="height: 0.625rem;"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="progress-bar__indicator-intent-primary"
        data-lens-element="progress-bar__indicator"
        data-lens-intent="primary"
        role="progressbar"
        style="width: 254.99999999999997%;"
      />
    </div>
  </div>
</div>
`;

exports[`<MessageBox/> should test the close button uncontrolled state 1`] = `
<div>
  <div
    id="lens-ui-portal-modal-root"
  />
  <div
    class="messageBox message-box--intent-primary"
    data-lens-element="message-box"
    data-testid="my_testing_id"
  >
    <div
      class="messageBoxTitle"
      data-lens-element="message-box__title"
    >
      my_title
    </div>
    <button
      class="messageBoxCloseButton message-box--intent-primary__close-button"
      data-lens-element="message-box__close-button"
    >
      <svg
        class="icon"
        data-lens-element="icon"
        fill="currentColor"
        height="0.8rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="0.8rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm3.354 4.646L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 1 1 .708-.708z"
        />
      </svg>
    </button>
    <div>
      <div
        class="messageBoxContent"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageBox/> should test the close button uncontrolled state 2`] = `
<div>
  <div
    id="lens-ui-portal-modal-root"
  />
  <div
    class="messageBox messageBoxHidden message-box--intent-primary"
    data-lens-element="message-box"
    data-testid="my_testing_id"
  >
    <div
      class="messageBoxTitle"
      data-lens-element="message-box__title"
    >
      my_title
    </div>
    <button
      class="messageBoxCloseButton message-box--intent-primary__close-button"
      data-lens-element="message-box__close-button"
    >
      <svg
        class="icon"
        data-lens-element="icon"
        fill="currentColor"
        height="0.8rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="0.8rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm3.354 4.646L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 1 1 .708-.708z"
        />
      </svg>
    </button>
    <div>
      <div
        class="messageBoxContent"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;
